### ISSUE TITLE
ClassVar implementation

### ISSUE THREAD

**Issue by miedzinski** [python/mypy#2771]
Currently `ClassVar` isn't implemented. I have opened python/typeshed#888 and submitted python/typeshed#889 (I hope it's enough). Once it's get merged mypy should implement this.

I would like to tackle this. I have started implementing this with `ClassVar` definition from PR I had submitted to typeshed and so far mypy is able to treat `ClassVar[T]` as `T`, which is expected. Of course assignment semantics are main purpose of `ClassVar` and I plan to implement them.

By the way, should mypy treat objects of this type as immutable, that is report errors on operations like calling `append` on `List`? AFAIK mypy doesn't have any concept of mutability.

**Comment by miedzinski** [comment]
I have some questions about class inheritance. Consider this:

```python
from typing import ClassVar


class A:
    x: ClassVar[str] = 'A'


class B(A):
    pass


A().x = 'A'  # clearly an error
B().x = 'b'  # what about this?
B.x = 'B'    # or this?


class C(A):
    x: str = 'C'


C().x = 'c'
```

I feel that assigning to instance of subclass should be an error, but I have no clue about subclass itself. I am not sure about redefining attribute to bare type in subclass - technically it is OK in terms of types, but what about semantics? If we want to check against any kind of shadowing class attribute then all these statements should be errors.

**Comment by ilevkivskyi** [comment]
I think it should be like this:
```python
A().x = 'A'  # clearly an error
B().x = 'b'  # also error
B.x = 'B'    # this is OK, also reading from B.x is OK

class C(A):
    x: str = 'C' # error

C().x = 'c' # error
```
The last two errors are because we should not allow to override classmethods with instance methods, see #1867. I think, for consistency the same should apply to variables.
